linsolve/tests/linsolve_test.py

```diff
@@ -320,6 +320,18 @@ def test_no_abs_phs_solve(self):
         self.assertAlmostEqual(np.angle(x), 0.)
         self.assertAlmostEqual(np.angle(y), 0.)
         self.assertAlmostEqual(np.angle(z), 0.)
+    def test_dtype(self):
+        for dtype in (np.float32, np.float64, np.complex64, np.complex128):
```

Comment:
Do you want to add a check for when things are other types? Like regular python complex or regular python float?

Reply:
I think you're looking at the wrong unit test for this.  TestLinSolve.test_infer_dtypes is what checks all the python data type stuff.